base/Dockerfile

```diff
@@ -25,4 +24,25 @@ RUN [ "$JITSI_RELEASE" = "unstable" ] && \
     apt-cleanup || \
     true
 
-ENTRYPOINT [ "/init" ]
+# supervisor installation &&
+# create directory for child images to store configuration in
+RUN apt-dpkg-wrap apt-get update && \
+    apt-dpkg-wrap apt-get -y install supervisor && \
+    mkdir -p /var/log/supervisor
+
+# fix permissions for timezone initialization
+RUN chgrp -R 0 /usr/share/zoneinfo && \
```

Comment:
I'd love some explanations about this because, I don't know how openstack behaves with docker so I'm probably missing some context here.

Reply:
Big explainer [here](https://cloud.redhat.com/blog/a-guide-to-openshift-and-uids) enlightened me, plus some tips from @benbz

From what I understand, GID 0 has no special treatment unlike UID 0 at a kernel level, meaning it's More Secure™ for some definitions. In any random Openshift cluster the default user environment will look something like this:

/etc/passwd
```
-- snip --
1001500000:x:1001500000:0:1001500000 user:/:/sbin/nologin
```

`whoami`
```
$ whoami
1001500000
```

Since the user is part of GID 0, it can read/write/run root files but it's not technically 'root' at the kernel level. Since we're not doing anything that would require root level kernel access in this context.

In Kubernetes by default unless specified in a securityContext in the pod spec the pod will run the container as root. Openshift is far more strict about what's running from a user/group perspective given the heavy SELinux involvement and by default will launch a container with a random (at least from the container's run environment perspective) UID and GID that's managed by Openshift.

This doesn't cause problems in well-formed containers that make no assumptions about users inside the container, so situations like here many of the Jitsi containers were creating users inside each container and modifying permissions work in 'stock k8s' because it's not strictly concerned but Openshift disallows setuid() preventing all that user switching logic from ever happening and failing the container hard.

The first layer of the onion was replacing s6-overlay with something that could run in userland, since s6-overlay was hard-coded to setuid(0) before starting anything which is impossible in Openshift without explicitly allowing the container to run with an 'anyuid' special role in Openshift, which was not a reasonable work around for those with strict security concerns and practices.

The second layer of the onion after that was owning all of the files that needed to be read/modified/run with group 0, since our main init process is no longer root and doesn't get special access to system files like `/var/run` without a little finessing. Since this image is based on the `debian` upstream image it brings along all the directory permission conventions that come with that including some of these directories not being writable by GID 0 even. So in this specific scenario we're changing the ownership of /usr/share/zoneinfo and making it read/write by group if it isn't already so that it can be reconfigured at runtime.